tests/testcases/030_check-network.yml

```diff
@@ -54,7 +54,28 @@
     delay: 5
 
   - name: Run 2 busybox pods in test ns
-    command: "{{ bin_dir }}/kubectl run {{ item }} --image={{ test_image_repo }}:{{ test_image_tag }} --namespace test --command -- tail -f /dev/null"
+    shell:
+      cmd: |
+        cat <<EOF | {{ bin_dir }}/kubectl apply -f -
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: {{ item }}
+          namespace: test
+        spec:
+          containers:
+          - name: busybox
+            image: {{ test_image_repo }}:{{ test_image_tag }}
+            command: ['tail', '-f', '/dev/null']
+            securityContext:
```

Comment:
To specify securityContext, this replaces `kubectl run` with `kubectl apply` because `kubect run` command doesn't support the details of securityContext.

Reply:
Still facing another issue after the above change:
```
TASK [Ping between pods is working] ********************************************
task path: /builds/kargo-ci/kubernetes-sigs-kubespray/tests/testcases/030_check-network.yml:155
Wednesday 12 October 2022  05:45:25 +0000 (0:00:00.069)       0:00:10.533 ***** 
failed: [instance-1] (item=['busybox1', '10.233.87.131']) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/usr/local/bin/kubectl", "-n", "test", "exec", "busybox1", "--", "ping", "-c", "4", "10.233.87.131"], "delta": "0:00:00.158110", "end": "2022-10-12 05:45:26.143461", "item": ["busybox1", "10.233.87.131"], "msg": "non-zero return code", "rc": 1, "start": "2022-10-12 05:45:25.985351", "stderr": "ping: permission denied (are you root?)\ncommand terminated with exit code 1", "stderr_lines": ["ping: permission denied (are you root?)", "command terminated with exit code 1"], "stdout": "PING 10.233.87.131 (10.233.87.131): 56 data bytes", "stdout_lines": ["PING 10.233.87.131 (10.233.87.131): 56 data bytes"]}
failed: [instance-1] (item=['busybox1', '10.233.87.132']) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/usr/local/bin/kubectl", "-n", "test", "exec", "busybox1", "--", "ping", "-c", "4", "10.233.87.132"], "delta": "0:00:00.168313", "end": "2022-10-12 05:45:26.549309", "item": ["busybox1", "10.233.87.132"], "msg": "non-zero return code", "rc": 1, "start": "2022-10-12 05:45:26.380996", "stderr": "ping: permission denied (are you root?)\ncommand terminated with exit code 1", "stderr_lines": ["ping: permission denied (are you root?)", "command terminated with exit code 1"], "stdout": "PING 10.233.87.132 (10.233.87.132): 56 data bytes", "stdout_lines": ["PING 10.233.87.132 (10.233.87.132): 56 data bytes"]}
failed: [instance-1] (item=['busybox2', '10.233.87.131']) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/usr/local/bin/kubectl", "-n", "test", "exec", "busybox2", "--", "ping", "-c", "4", "10.233.87.131"], "delta": "0:00:00.141854", "end": "2022-10-12 05:45:26.930681", "item": ["busybox2", "10.233.87.131"], "msg": "non-zero return code", "rc": 1, "start": "2022-10-12 05:45:26.7[8882](https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/3159916632#L8882)7", "stderr": "ping: permission denied (are you root?)\ncommand terminated with exit code 1", "stderr_lines": ["ping: permission denied (are you root?)", "command terminated with exit code 1"], "stdout": "PING 10.233.87.131 (10.233.87.131): 56 data bytes", "stdout_lines": ["PING 10.233.87.131 (10.233.87.131): 56 data bytes"]}
failed: [instance-1] (item=['busybox2', '10.233.87.132']) => {"ansible_loop_var": "item", "changed": true, "cmd": ["/usr/local/bin/kubectl", "-n", "test", "exec", "busybox2", "--", "ping", "-c", "4", "10.233.87.132"], "delta": "0:00:00.182038", "end": "2022-10-12 05:45:27.365910", "item": ["busybox2", "10.233.87.132"], "msg": "non-zero return code", "rc": 1, "start": "2022-10-12 05:45:27.183872", "stderr": "ping: permission denied (are you root?)\ncommand terminated with exit code 1", "stderr_lines": ["ping: permission denied (are you root?)", "command terminated with exit code 1"], "stdout": "PING 10.233.87.132 (10.233.87.132): 56 data bytes", "stdout_lines": ["PING 10.233.87.132 (10.233.87.132): 56 data bytes"]}
```

Reply:
The busybox image is not good for non-root testing because that requires root for ping.
One workaround is to add CAP_NET_RAW capability like https://github.com/kubernetes/kubernetes/pull/95321/files